src/language-js/print/binaryish.js

```diff
@@ -289,11 +318,15 @@ function printBinaryishExpressions(
       parent.type !== node.type &&
       node.left.type !== node.type &&
       node.right.type !== node.type);
+  if (shouldGroup) {
+    right = group(right, { shouldBreak });
```

Comment:
This line came from the inline expression `shouldGroup ? group(right, { shouldBreak }) : right`. Now the possibly-grouped `right` is used by both branches' `.push()` calls below.